bqskit/qis/state/state.py

```diff
@@ -166,6 +166,43 @@ def is_qudit_only(self, radix: int) -> bool:
                 states.
         """
         return all([r == radix for r in self.radixes])
+    
+    def get_sub_state(self, location: CircuitLocationLike) -> StateVector:
+        """
+        Return the StateVector corresponding to the specified location
+        in the Circuit.
```

Comment:
I would drop the "in the Circuit" and make this one line. I am also curious how does renormalization happen? Can you provide either more detail about what this does mathematically, provide a reference, or add an example (or two)? Ideally, all of the above.